instrumentation/spring/spring-integration-4.1/javaagent/src/test/java/SpringIntegrationAndRabbitTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import static io.opentelemetry.instrumentation.testing.GlobalTraceUtil.runWithSpan;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions;
+import io.opentelemetry.semconv.NetworkAttributes;
+import io.opentelemetry.semconv.incubating.MessagingIncubatingAttributes;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class SpringIntegrationAndRabbitTest extends AbstractRabbitProducerConsumerTest {
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @BeforeEach
+  public void setupSpec() {
+    startRabbit(null);
+  }
+
+  @AfterEach
+  public void cleanupSpec() {
+    stopRabbit();
+  }
+
+  @Test
+  public void shouldCooperateWithExistingRabbitMQInstrumentation() {
+    runWithSpan("parent", () -> producerContext.getBean("producer", Runnable.class).run());
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent"),
+                span -> span.hasName("producer").hasParent(trace.getSpan(0)),
+                span ->
+                    span.hasName("exchange.declare")
+                        .hasParent(trace.getSpan(1))
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            OpenTelemetryAssertions.satisfies(
```

Comment:
we could use static imports for these
```suggestion
                            satisfies(
```